Milestone4 requirements:

11galpid opened the pull request:
-Automatic launch of app
-Basic Usage & launch of app in README

11galpid committed:
Milestone4 requirements:
-Automatic launch of app
-Basic Usage for launch of app

ifireball commented on Vagrantfile:
```diff
@@ -12,6 +12,9 @@ Vagrant.configure("2") do |config|
   config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
 
+  # Shortcuts used later inside dev vm
+  config.vm.provision "file", source: "aliases.sh", destination: "~/.aliases.sh"
```
This sets up the aliases for the root account and for the duration of the `aliases.sh` run - Probably no what you want. To properly set aliases you must have them defined from a shell startup file such as `.bashrc` (You can have `.bashrc` source another script if you want)

11galpid replied:
It set's up fpr the vagrant user, not root. Verified.
However maybe implicitly stating /home/vagrant is more appropriate. Will change

ifireball replied:
Oh I think I misunderstood what you're doing here - so this is installing the file, not running it? You may want to alter your comment above to make that clearer.

Also, maybe just source the file from `/vagrant` instead of explicitly copying it to another location?

ifireball commented on README.md:
```diff
@@ -68,4 +68,18 @@ Guy Itzhaki, Tarel Madar, Gal Lapid, Alon Weissfeld, Omri Rosner
-4. Run 'vagrant up'. Once it finished, you're all set!
\ No newline at end of file
+4. Run 'vagrant up'. Once it finished, you're all set!
+
```
The instructions below should not be needed, `vagrant up` and `vgrant halt` should be the only command that other developers need - unless thry need to connect to the DB directly or something like that.

If people need to run commands like below, you are not meeting the 4th milestone requirements.

ifireball commented on init_scripts/init-user.sh:
```diff
@@ -1,11 +1,19 @@
+# Also, we specify the version to avoid possible inconsistency
+# with future releases, and --user to reduce global fingerprint
+# on the system as a whole.
+python -m pip install -I pipenv==2020.8.13 --user
```
you should probably put `python3` here to ensure you're always running the right python version

ifireball commented on init_scripts/init-user.sh:
```diff
@@ -1,11 +1,19 @@
+python -m pip install -I pipenv==2020.8.13 --user
+cd /vagrant/sellotape && pipenv sync
+
+# Start sellotape django webserver
```
I think you should be running migrations here as well

11galpid replied:
Right, will be added

ifireball reviewed (commented):
The PR title assumes internal knowledge about milestones - it could be better.

I see some things in this PR that I was not expecting, and some things seem to be missing. One example is that I'm not seeing automated setup of the super user.

11galpid committed:
Infrastructure enhancements
-Automatic superuser creation using default credentials
-Changes in the README (removal of manual steps no longer needed)
-Automatic migration upon vigrant up
-.aliases.sh is no longer copied, using from /vagrant shared folder

ifireball commented on init_scripts/init-user.sh:
```diff
@@ -1,11 +1,24 @@
+# Create required models
+pipenv run python3 sellotape_dj/manage.py migrate
+# Create default super user
+export DJANGO_SUPERUSER_PASSWORD='123456'
```
See [here](https://github.com/ifireball/knowbali-containers-in-depth/commit/48b00daaab72352b6a2786f2c0d629045d3c11a1) for how to use a migration to create the super user - not critical for now but you may want to do it later.

ifireball reviewed (approved):
generally LGTM

11galpid committed:
Merge branch 'master' into dev-milestone4
